haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -1,23 +1,393 @@
+    HasMintDistribution (..),
+
+    -- * Transaction fee distribution
+    TransactionFeeDistribution (..),
+    -- |Typeclass for structures that contain a 'TransactionFeeDistribution'.
+    HasTransactionFeeDistribution (..),
+
+    -- * GAS rewards
+
+    -- |Versioning for the 'GASRewards' parameters.
+    --
+    -- * 'GASRewardsVersion0' ('ChainParametersV0', 'ChainParametersV1'): supports GAS reward for
+    --   including finalization proofs.
+    -- * 'GASRewardsVersion1' ('ChainParametersV1'): does not support GAS reward for including
+    --   finalization proofs.
+    GASRewardsVersion (..),
+    -- |Singleton type associated with 'GASRewardsVersion'.
+    SGASRewardsVersion (..),
+    -- |The GAS rewards version for a chain parameters version.
+    gasRewardsVersionFor,
+    -- |The GAS rewards version for a chain parameters version (types).
+    GasRewardsVersionFor,
+    -- |The GAS rewards version for a chain parameters version (singletons).
+    sGasRewardsVersionFor,
+    IsGASRewardsVersion,
+    withIsGASRewardsVersionFor,
+    -- |Whether a 'GASRewardsVersion' supports GAS rewards for finalization proofs.
+    supportsGASFinalizationProof,
+    -- |Whether a 'GASRewardsVersion' supports GAS rewards for finalization proofs (type level).
+    SupportsGASFinalizationProof,
+    -- |Whether a 'GASRewardsVersion' supports GAS rewards for finalization proofs (singletons).
+    sSupportsGASFinalizationProof,
+    withSupportsGASFinalizationProof,
+    GASRewards (..),
+    -- |Typeclass for structures that contain a 'GASRewards'.
+    HasGASRewards (..),
+
+    -- * Reward parameters
+    RewardParameters (..),
+    -- |Typeclass for structures that contain a 'RewardParameters'.
+    HasRewardParameters (..),
+
+    -- * Exchange rates
+    ExchangeRates (..),
+    makeExchangeRates,
+    HasExchangeRates (..),
+
+    -- * Cooldown parameters
+
+    -- |Versioning for the 'CooldownParameters'' structure.
+    --
+    -- * 'CooldownParametersVersion0' ('ChainParametersV0'): baker cooldown specified in 'Epoch's.
+    -- * 'CooldownParametersVersion1' ('ChainParametersV1', 'ChainParametersV2'): baker and
+    --   delegator cooldowns specified in seconds.
+    CooldownParametersVersion (..),
+    -- |Singleton type associated with 'CooldownParametersVersion'.
+    SCooldownParametersVersion (..),
+    -- |The cooldown parameters version for a chain parameters version.
+    cooldownParametersVersionFor,
+    -- |The cooldown parameters version for a chain parameters version (types).
+    CooldownParametersVersionFor,
+    -- |The cooldown parameters version for a chain parameters version (singletons).
+    sCooldownParametersVersionFor,
+    IsCooldownParametersVersion,
+    withIsCooldownParametersVersionFor,
+    CooldownParameters' (..),
+    CooldownParameters,
+    cpBakerExtraCooldownEpochs,
+    cpPoolOwnerCooldown,
+    cpDelegatorCooldown,
+    putCooldownParameters,
+    getCooldownParameters,
```

Comment:
Why are these not documented?

Reply:
There are documentation at the actual definitions. Is this not adequate? It seems to me that duplicating the stubs of documentation will just lead to an unnecessary overhead when maintaining the code base.

Reply:
I have put a comment at top stating that TH definitions are documented in the header of the module while the regular definitions are simply documented where they are defined.